snuba/clusters/cluster.py

```diff
@@ -399,11 +395,46 @@ def _get_storage_set_cluster_map() -> Dict[StorageSetKey, ClickhouseCluster]:
     return _STORAGE_SET_CLUSTER_MAP
 
 
+def _build_partitioned_cluster(cluster: Mapping[str, Any]) -> ClickhouseCluster:
+
+    return ClickhouseCluster(
+        host=cluster["host"],
+        port=cluster["port"],
+        user=cluster.get("user", "default"),
+        password=cluster.get("password", ""),
+        database=cluster.get("database", "default"),
+        http_port=cluster["http_port"],
+        storage_sets={storage_tuple[0] for storage_tuple in cluster["storage_sets"]},
+        single_node=cluster["single_node"],
+        cluster_name=cluster["cluster_name"] if "cluster_name" in cluster else None,
+        distributed_cluster_name=cluster["distributed_cluster_name"]
+        if "distributed_cluster_name" in cluster
+        else None,
+        cache_partition_id=cluster.get("cache_partition_id"),
+        query_settings_prefix=cluster.get("query_settings_prefix"),
+    )
+
+
+def _get_partitioned_storage_set_cluster_map() -> Dict[
+    Tuple[StorageSetKey, int], ClickhouseCluster
+]:
+    _PARTITIONED_STORAGE_SET_CLUSTER_MAP = {}
```

Comment:
We should probably only build this map once (e.g `_PARTITIONED_STORAGE_SET_CLUSTER_MAP` should be a global and we only assign it if it's equal to empty map at the start of the function call).